exercises/concept/making-the-grade/.meta/src/reference/java/MakingTheGrade.java

```diff
@@ -0,0 +1,56 @@
+import java.util.List;
+import java.util.ArrayList;
+
+class MakingTheGrade {
+
+    static int countFailedStudents(List<Integer> studentScores) {
+        int count = 0;
+
+        for (int score : studentScores) {
+            if (score <= 40) {
+                count++;
+            }
+        }
+
+        return count;
+    }
+
+    static int getAverageScoreOfStudents(List<Integer> studentScores, int numberOfStudents) {
+        int count = 0;
+        int sum = 0;
+
+        while (count < numberOfStudents) {
+            sum += studentScores.get(count);
+            count++;
+        }
```

Comment:
What makes this method a good candidate for a `while` loop? I could just as easily (and probably would) implement this using a `for` loop:

```java
int sum = 0;

for (int i = 0; i < numberOfStudents; i++) {
    sum += studentScores.get(i);
}

return sum / numberOfStudents;
```

Reply:
So my train of thought was that as I'm giving `numberOfStudents` as a parameter for the method and `while` is condition centric loop that would guide students to use `while` instead of `for` but yeah it can be solved the way you said as well. Do you have another ideas?